tron/core/jobrun.py

```diff
@@ -395,62 +382,44 @@ def get_run_by_index(self, index):
 
     def get_run_by_state_short_name(self, short_name):
         """Returns the most recent run which matches the state short name."""
-        return self._get_run_using(lambda r: r.state.short_name == short_name)
+        return next_or_none(r for r in self.runs if r.state.short_name == short_name)
 
     def get_newest(self, include_manual=True):
         """Returns the most recently created JobRun."""
-
-        def func(r):
-            return True if include_manual else not r.manual
-
-        return self._get_run_using(func)
+        return next_or_none(r for r in self.runs if include_manual or not r.manual)
 
     def get_pending(self):
         """Return the job runs that are queued or scheduled."""
-        return self._get_runs_using(lambda r: r.is_scheduled or r.is_queued)
+        return [r for r in self.runs if r.is_scheduled or r.is_queued]
```

Comment:
why not next_or_none(r for r in self.runs if r.is_scheduled or r.is_queued) ?

Reply:
`next_or_none` is for cases when we only need first element of the sequence, in this  case we need  the sequence.